components/modals/EditTaskModal.tsx

```diff
@@ -0,0 +1,147 @@
+  const [task, setTask] = useState(initialTask);
+  const [initialValues, setInitialValues] = useState<TaskFormData>({
+    notificationDaysOffset: DEFAULT_NOTIFICATION_DAYS_OFFSET,
+  });
+  const [shouldDisableSaveButton, setShouldDisableSaveButton] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [taskFormData, setTaskFormData] = useState<Nullable<TaskFormData>>(null);
+
+  useEffect(() => {
+    const taskDueDate = isValidDate(task.dueDate) ? new Date(task.dueDate) : undefined;
+    const taskDueDateMoment = taskDueDate !== undefined ? moment(taskDueDate) : undefined;
+    const taskNotificationDateTime =
+      task.notificationDateTime !== null && isValidDate(task.notificationDateTime)
+        ? new Date(task.notificationDateTime)
+        : undefined;
+    const taskNotificationDateTimeMoment =
+      taskNotificationDateTime !== undefined ? moment(taskNotificationDateTime) : undefined;
+    const calculatedNotificationDaysOffset =
+      taskDueDate && taskNotificationDateTime
+        ? calculateDaysOffset(taskNotificationDateTime, taskDueDate)
+        : DEFAULT_NOTIFICATION_DAYS_OFFSET;
+    const notificationDateTimeType =
+      taskNotificationDateTime === undefined
+        ? NotificationDateTimeType.NONE
+        : calculatedNotificationDaysOffset === 0
+        ? NotificationDateTimeType.DAY_OF_EVENT
+        : calculatedNotificationDaysOffset > 0
+        ? NotificationDateTimeType.DAYS_BEFORE
+        : NotificationDateTimeType.DAYS_AFTER;
+
+    setInitialValues({
+      title: task.title,
+      dueDate: taskDueDateMoment,
+      notificationDaysOffset: Math.abs(calculatedNotificationDaysOffset),
+      notificationDateTimeType: notificationDateTimeType,
+      notificationDate: taskNotificationDateTimeMoment,
+      notificationTime: taskNotificationDateTimeMoment ?? DEFAULT_NOTIFICATION_TIME,
+      isDone: task.isDone,
+    });
+  }, [task]);
+
+  useEffect(() => {
+    if (taskFormData !== null) {
+      handleSubmit(taskFormData);
+    }
+  }, [taskFormData]);
+
+  const onCancel = () => {
+    setSelectedTask(null);
+  };
+
+  const onSubmit = () => {
+    setIsSubmitting(true);
+  };
+
+  const onDelete = () => {
+    confirm({
+      title: 'Are you sure about deleting this task?',
+      icon: <ExclamationCircleOutlined />,
+      content: 'This action is irreversible',
+      onOk() {
+        handleDelete();
+      },
+    });
+  };
+
+  const handleSubmit = (values: TaskFormData) => {
+    const taskPatchData: TaskPatchData = {
+      title: values.title,
+      dueDate: values.dueDate?.toISOString(),
+      notificationDateTime: getNotificationDateTime(values),
+      isDone: values.isDone,
+    };
+
+    tasksApi
+      .editTask(applicationId, task.id, taskPatchData)
+      .then((value) => {
+        const updatedTask = value.payload as TaskData;
+        setTask(updatedTask);
+        setShouldFetchData(true);
```

Comment:
Instead of doing this, is it better to just mutate instead?

https://swr.vercel.app/docs/mutation

Refer to `pages/applications/create.tsx` of HY PR line 92.

Reply:
This is just telling the parent component to mutate, the parent component is still using mutate. Unless you want me to pass the mutate function down to the child? Ref to `/[applicationId].tsx` line 72.

If you're talking about mutating task, Task does not have a GET endpoint.